### PR TITLE
Change uses of assert that are meant to persist with a different macro

### PR DESCRIPTION
**Pull request by Zeno-**
Let's get rid of the incorrect usages of assert()

### CODE REVIEW

**Comment by nerzhul** [comment]
What does this new assert change ?

**Comment by Zeno-** [comment]
@nerzhul nothing functionally different. Every single usage of the new macro needs to be checked to see if it's a properly used assert or if it was used as a replacement for error checking. If it was properly used then it will be replaced with the standard assert() (which of course are no-ops when NDEBUG is defined so therefore removed in release builds, unlike how things are currently)

**Comment by Zeno-** [comment]
I wish there were some more appropriate labels :/

File: src/client.cpp [hunk 1]
```diff
@@ -785,7 +788,9 @@ void Client::request_media(const std::vector<std::string> &file_requests)
 	std::ostringstream os(std::ios_base::binary);
 	writeU16(os, TOSERVER_REQUEST_MEDIA);
 	size_t file_requests_size = file_requests.size();
-	assert(file_requests_size <= 0xFFFF);
+
+	if (file_requests_size > 0xFFFF)
+		FATAL_ERROR("Unsupported number of file requests");
```
Comment (nerzhul) [inline src/client.cpp]: Why not use FATAL_ERROR_IF ?

File: src/client.cpp [hunk 2]
```diff
@@ -1295,7 +1299,7 @@ Inventory* Client::getInventory(const InventoryLocation &loc)
 	}
 	break;
 	default:
-		assert(0);
+		FATAL_ERROR("Invalid inventory locationt type.");
```
Comment (nerzhul) [inline src/client.cpp]: typo

File: src/client.cpp (hunk 1 above)
Comment (Zeno-) [inline src/client.cpp]: I probably hadn't created that macro when I wrote that :)

**Comment by Zeno-** [comment]
commit ced6d20295a8263757d57c02a07ffcb66688a163

**Pull request closed without merging**